src/Microsoft.AspNet.Razor.Runtime/TagHelpers/RestrictChildrenAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Restricts children of the <see cref="ITagHelper"/>'s element.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, Inherited = true, AllowMultiple = false)]
+    public class RestrictChildrenAttribute : Attribute
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="RestrictChildrenAttribute"/> class.
+        /// </summary>
+        /// <param name="tagName">The tag name of the HTML element to allow as children.</param>
+        /// <param name="tagNames">Additional tag names of HTML elements to restrict as children.</param>
+        public RestrictChildrenAttribute(string tagName, params string[] tagNames)
+        {
+            var concatenatedNames = new string[1 + tagNames.Length];
+            concatenatedNames[0] = tagName;
+
+            tagNames.CopyTo(concatenatedNames, 1);
+
+            ChildTagNames = concatenatedNames;
+        }
+
+        /// <summary>
+        /// HTML element tag names targeted by tag helpers allowed as children.
```

Comment:
Suggest `Get the names of elements allowed as children. Tag helpers must target all such elements.` and everything but `Get` in the parameter description.